cmake/import/nabla.h.in

```diff
@@ -1,160 +1,159 @@
-	device could not be created. */
-	core::smart_refctd_ptr<IrrlichtDevice> createDeviceEx(const SIrrlichtCreationParameters& parameters);
-
-
-	// THE FOLLOWING IS AN EMPTY LIST OF ALL SUB NAMESPACES
-	// EXISTING ONLY FOR THE DOCUMENTATION SOFTWARE DOXYGEN.
-
-	//! Basic classes such as vectors, planes, arrays, lists, and so on can be found in this namespace.
-	namespace core
-	{
-	}
-
-	//! This namespace provides interfaces for input/output: Reading and writing files, accessing zip archives, faking files ...
-	namespace io
-	{
-	}
-    
-    //! All asset loading and mutation is performed here: Loading and Saving Images, Models, Shaders, Mesh and Texture CPU manipulation ...
-    namespace asset
-    {
-    }
-
-	//! All scene management can be found in this namespace: scene graph, scene nodes, cameras, animation, etc...
-	namespace scene
-	{
-	}
-
-	//! The video namespace contains classes for accessing the graphics API
-	namespace video
-	{
-	}
-}
-
-/*! \file irrlicht.h
-	\brief Main header file of the irrlicht, the only file needed to include.
-*/
-
-#endif
-
+/* nabla.h -- interface of the 'Nabla Engine'
+
+  Copyright (C) 2019 - DevSH Graphics Programming Sp. z O.O.
+
+  This software is provided 'as-is', under the Apache 2.0 license,
+  without any express or implied warranty.  In no event will the authors
+  be held liable for any damages arising from the use of this software.
+
+  See LICENSE.md for full licensing information.
+
+  Please note that the IrrlichtBAW Engine is based in part on the work of others,
+  this means that if you use the IrrlichtBAW Engine in your product,
+  you must acknowledge somewhere in your documentation that you've used their code.
+  See README.md for all mentions of 3rd party software used.
+  
+  The file is generated by CMake,
+  to modify the code open ${NBL_ROOT_PATH}/cmake/import/nabla.h.in.
+*/
+
+#ifndef __NABLA_H_INCLUDED__
+#define __NABLA_H_INCLUDED__
+
+// core lib
+#include "nbl/core/core.h"
+
+// system lib (fibers, mutexes, file I/O operations) [DEPENDS: core]
+#include "nbl/system/system.h"
+// should we move "core/parallel" to "system/parallel"
+
+// asset lib (importing and exporting meshes, textures and shaders) [DEPENDS: system]
+#include "nbl/asset/asset.h"
+// ui lib (window set up, software blit, joysticks, multi-touch, keyboard, etc.) [DEPENDS: system]
+#include "nbl/ui/ui.h"
+
+// video lib (access to Graphics API, remote rendering, etc) [DEPENDS: asset, (optional) ui]
+#include "nbl/video/video.h"
+
+// scene lib (basic rendering, culling, scene graph etc.) [DEPENDS: video, ui]
+#include "nbl/scene/scene.h"
+
+#include "aabbox3d.h"
+#include "vector2d.h"
+#include "vector3d.h"
+#include "vectorSIMD.h"
+#include "line3d.h"
+#include "matrix4SIMD.h"
+#include "position2d.h"
+#include "quaternion.h"
+#include "rect.h"
+#include "dimension2d.h"
+#include "EDriverTypes.h"
+#include "ICameraSceneNode.h"
+#include "ICursorControl.h"
```

Comment:
is @sadiuk killing the cursor control @Crisspl ?

Reply:
yea, absolutely, it's never been used by us tbh

Reply:
well, "killing", we're just not touching it

Reply:
lets kill dead code, it makes me happy